python/amici/petab_import.py

```diff
@@ -73,25 +73,60 @@ def _add_global_parameter(sbml_model: libsbml.Model,
 
 
 def get_fixed_parameters(
-        sbml_model: 'libsbml.Model',
-        condition_df: Optional[pd.DataFrame] = None,
+        petab_problem: petab.Problem
 ) -> List[str]:
     """
     Determine, set and return fixed model parameters.
 
-    Parameters specified in `condition_df` are turned into constants.
+    Non-estimated parameters and parameters specified in the condition table
+    are turned into constants (unless they are overridden).
     Only global SBML parameters are considered. Local parameters are ignored.
 
-    :param condition_df:
-        PEtab condition table. If provided, the respective parameters
-        will be turned into AMICI constant parameters.
-
-    :param sbml_model:
-        libsbml.Model instance
+    :param petab_problem:
+        The PEtab problem instance
 
     :return:
         List of IDs of parameters which are to be considered constant.
     """
+    # initial concentrations for species or initial compartment sizes in
+    # condition table will need to be turned into fixed parameters
+
+    # if there is no initial assignment for that species, we'd need
+    # to create one. to avoid any naming collision right away, we don't
+    # allow that for now
+
+    # we can't handle them yet
+    compartments = [
+        col for col in petab_problem.condition_df
+        if petab_problem.sbml_model.getCompartment(col) is not None
+    ]
+    if compartments:
+        raise NotImplementedError("Can't handle initial compartment sizes "
+                                  "at the moment. Consider creating an "
+                                  f"initial assignment for {compartments}")
+
+    # if we have a parameter table, all parameters that are allowed to be
+    #  listed in the parameter table, but are not marked as estimated, can be
+    #  turned in to AMICI constants
+    # due to legacy API, we might not always have a parameter table, though
+    fixed_parameters = set()
+    if petab_problem.parameter_df is not None:
+        all_parameters = get_valid_parameters_for_parameter_table(
+            model=petab_problem.model,
+            condition_df=petab_problem.condition_df,
+            observable_df=petab_problem.observable_df
+            if petab_problem.observable_df is not None
```

Comment:
wouldn't it be better to have some default handling in `get_valid_parameters_for_parameter_table`?

Reply:
That will require a new `libpetab` release, so I will leave that in amici for now.